### PR TITLE
rtos: abstraction: partition RTOS logic into separate directories

### PR DESCRIPTION
**Pull request by lgirdwood**
In preparation for further RTOS abstraction, move the xtos and zephyr specific headers to new top level rtos directories to enable:

1) Partitioning of code by RTOS type e.g. xtos, Zephyr.
2) Partitioning of RTOS and non RTOS logic and APIs

This PR reuses the existing zephyr top level directory and creates a new xtos top level directory to store RTOS specific files, logic and wrappers. Other RTOSes like FreeRTOS or CMSIS support would be added in the same way.

This PR makes no runtime functional change, but changes some RTOS specific headers to be included with a new directory. i.e.

```c
#include <sof/spinlock.h>
```
now becomes
```c
#include <rtos/spinlock.h>
```
As this now maps to the correct RTOS and is no longer using the ```sof``` prefix which is for common code.

This PR also adds a high level Kconfig for developers (disabled by default) so that they can easily switch to fully native Zephyr headers for development testing.

Signed-off-by: Liam Girdwood <liam.r.girdwood@linux.intel.com>

### CODE REVIEW

**Review by dbaluta — APPROVED** [review]

File: zephyr/CMakeLists.txt [hunk 1]
```diff
@@ -88,6 +88,10 @@ target_include_directories(SOF INTERFACE ../zephyr/include)
 target_include_directories(SOF INTERFACE ${SOF_SRC_PATH}/include)
 target_include_directories(SOF INTERFACE ${SOF_SRC_PATH}/arch/${ARCH}/include)
 
+# TODO: The xtos-wrapper include PATH will eventually become optional based on the
+# RTOS selection. It's manadatory atm until the native Zephyr integration is ready.
+target_include_directories(SOF INTERFACE ${SOF_SRC_PATH}/arch/xtos-wrapper/include)
```
Comment (lyakh) [inline zephyr/CMakeLists.txt:93]: hm, is this needed even if `CONFIG_ZEPHYR_NATIVE_DRIVERS` is used? If not, maybe make it conditional on this, if yes - maybe keep the headers, that Zephyr builds still need at their current locations and only move them over one by one as we stop using them in Zephyr?

Comment (lgirdwood) [reply zephyr/CMakeLists.txt:93]: yep, that the plan and why it's a TODO. Its still WIP atm.

**Review by dbaluta — APPROVED** [review]

File: src/include/Readme.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+ ** RTOS Header Decoupling **
+
+ Due to the mixing of RTOS, driver and library headers at the top level include
+ directory it was necessary to create rtos specific header directories i.e.
+
+```
+src/include/rtos-xtos
+src/include/rtos-zephyr
+```
```
Comment (lgirdwood) [inline src/include/Readme.md:30]: This is out of date, I need to update the description.

**Review by andyross — APPROVED** [review]
This all makes sense to me.  Unfortunately it looks likely to collide badly with some terrible (but thematically similar) hackery I've been doing to get SOF to build against the native_posix architecture for an internal fuzzing requirement.  There the glitch is "xtensa assumptions in headers" and not "xthal/zephyr assumptions", but the theme is similar.

**Comment by lgirdwood** [comment]
SOFCI TEST

**Review by RanderWang — APPROVED** [review]

**Comment by lgirdwood** [comment]
License server issues on retired platforms. No need to rebuild.

**Pull request merged**